bindings/experimental/distrdf/python/DistRDF/ComputationGraphGenerator.py

```diff
@@ -143,3 +151,68 @@ def generate_computation_graph(node_cpp, range_id, node_py=None):
             return return_vals
 
         return generate_computation_graph
+
+    def get_callable_optimized(self):
+        """
+        Converts a given graph into a callable and returns the same.
+        The callable is optimized to execute the graph with compiled C++
+        performance.
+
+        Returns:
+            function: The callable that takes in a PyROOT RDataFrame object
+            and executes all operations from the DistRDF graph
+            on it, recursively.
+        """
+        # Prune the graph to check user references
+        self.headnode.graph_prune()
+
+        def run_computation_graph(rdf_node, range_id):
+            """
+            The callable that traverses the DistRDF graph nodes, generates the
+            code to create the same graph in C++, compiles it and runs it.
+            This function triggers the event loop via the CppWorkflow class.
+
+            Args:
+                rdf_node (ROOT.RDF.RNode): The main RDataFrame node on which
+                    the graph will be executed.
+                range_id (int): Id of the current range. Needed to assign a name
+                    to a partial Snapshot output file.
+
+            Returns:
+                tuple: the first element is the list of results of the actions
+                    in the C++ workflow, the second element is the list of
+                    result types corresponding to those actions.
+            """
+
+            py_headnode = self.headnode
+            cpp_workflow = CppWorkflow()
+            parent_idx = 0
+
+            # Recurse over children nodes
+            for py_node in py_headnode.children:
+                explore_graph(py_node, cpp_workflow, range_id, parent_idx)
```

Comment:
Perhaps yes, since the calls to `add_node` receive a `parent_idx`, which implies some knowledge about the existence of a graph. Otherwise it would just be fine to traverse the graph outside and add new operations individually.

So I'll move this graph traversal to `__init__` of `CppWorkflow` - still I'll flag the individual `add_node`, `add_lambda`, etc. operations as public, just in case we need to do individual additions later (e.g. from local RDF).